apps/beeswax/src/beeswax/server/dbms.py

```diff
@@ -34,19 +36,22 @@
 
 from beeswax import hive_site
 from beeswax.conf import HIVE_SERVER_HOST, HIVE_SERVER_PORT, HIVE_METASTORE_HOST, HIVE_METASTORE_PORT, LIST_PARTITIONS_LIMIT, SERVER_CONN_TIMEOUT, \
-  AUTH_USERNAME, AUTH_PASSWORD, APPLY_NATURAL_SORT_MAX, QUERY_PARTITIONS_LIMIT
+  AUTH_USERNAME, AUTH_PASSWORD, APPLY_NATURAL_SORT_MAX, QUERY_PARTITIONS_LIMIT, HIVE_DISCOVERY_HIVESERVER2_ZNODE, \
+  HIVE_DISCOVERY_HS2, HIVE_DISCOVERY_LLAP, HIVE_DISCOVERY_LLAP_HA, HIVE_DISCOVERY_LLAP_ZNODE, CACHE_TIMEOUT, \
+  LLAP_SERVER_HOST, LLAP_SERVER_PORT
 from beeswax.common import apply_natural_sort
 from beeswax.design import hql_query
 from beeswax.hive_site import hiveserver2_use_ssl
 from beeswax.models import QueryHistory, QUERY_TYPES
-
+from libzookeeper import conf as libzookeeper_conf
 
 LOG = logging.getLogger(__name__)
 
 
 DBMS_CACHE = {}
 DBMS_CACHE_LOCK = threading.Lock()
-
+cache = caches['hive_discovery']
+cache.clear()
```

Comment:
nit: really needed? (on startup should be empty anyway)

Reply:
the file cache persists across restarts (the only option to expire is time based).  This presents a problem for hiveserver2 since we aren't expiring that cache entry and there isn't a mechanism yet to expire the cache on a failure.

Reply:
Interesting, maybe add a little comment there?